core/ui_mixins/quick_panel.py

```diff
@@ -168,7 +162,7 @@ def __init__(self, items, on_done, **kwargs):
         self.on_done = on_done
         for option in ['flags', 'selected_index', 'on_highlight',
                        'limit', 'next_message', ]:
-            if option in kwargs:
+            if option in kwargs and kwargs[option]:
```

Comment:
Oh I think I see now.. we do this so that show_log_panel can accept `on_highlight`?

I truly found it a bit hard to follow, maybe we need to add some comments or docstrings, at least so we don't forget why it's done in the future?